sdk/core/azure-core/inc/azure/core/internal/client_options.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Base type for all client option types, exposes various common client options like Retry
+ * and Transport.
+ */
+
+#pragma once
+
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/policy.hpp"
+
+#include <memory>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Internal {
+
+  /**
+   * @brief Represents a position of the policy in the pipeline.
+   *
+   */
+  enum class HttpPipelinePosition
+  {
+    /**
+     * @brief The policy would be invoked once per pipeline invocation (service call).
+     *
+     */
+    PerCall,
+    /**
+     * @brief The policy would be invoked every time request is retried.
+     *
+     */
+    PerRetry
+  };
+
+  /**
+   * @brief  Base type for all client option types, exposes various common client options like Retry
+   * and Transport.
+   *
+   */
+  class ClientOptions {
+  private:
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> m_perOperationPolicies;
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> m_perRetryPolicies;
+
+  public:
+    /**
+     * @brief Move each policy from \p options into the new instance.
+     *
+     * @param options
+     */
+    explicit ClientOptions(ClientOptions&& options)
```

Comment:
So when you make a copy of `ClientOptons`, only policies are preserved, all other fields are just abandoned?

Reply:
Yes, to create the pipeline, Core only cares about the client options in the base class.
All other fields are used by each service sdk to create the list of policies that are passed to core to be added to the pipeline